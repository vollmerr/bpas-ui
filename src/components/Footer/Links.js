import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import theme from '../theme';

import Version from './Version';

const urls = [
  {name: 'Conditions of Use', to: 'http://www.ca.gov/Use'},
  {name: 'Privacy Policy', to: 'http://www.ca.gov/Privacy'},
  {name: 'Accessibility', to: 'http://www.ca.gov/Accessibility'},
  {name: 'Contact Us', to: '/help'},
];

const Wrapper = styled.div`
  display: flex;
  flex: 6;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  margin: 0;
  padding: 0;
`;

const StyledLink = styled(Link)`
  color: ${theme.color.primaryInverse};
  white-space: nowrap;
  min-width: 150px;
  display: flex;
  transition: all 0.25s;

  &:hover,
  &:focus {
    color: ${theme.color.second};
  }
`;

const StyledAnchor = StyledLink.withComponent('a');

function link(url) {
  if(url.to.match(/^http/)) {
    return <StyledAnchor href={url.to} key={url.name}>{url.name}</StyledAnchor>;
  } else {
    return <StyledLink to={url.to} key={url.name}>{url.name}</StyledLink>;
  }
}

function Links() {
  return (
    <Wrapper>
      {urls.map(url => link(url))}
      <Version />
    </Wrapper>
  );
}

export default Links;
