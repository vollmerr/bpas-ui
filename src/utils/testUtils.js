import React from 'react';
import { StaticRouter } from 'react-router';

export function testRouter(component) {
  const context = {};
  const location = './';

  return (
    <StaticRouter location={location} context={context}>
      {component}
    </StaticRouter>
  );
}
