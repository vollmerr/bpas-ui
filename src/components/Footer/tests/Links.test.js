import React from 'react';
import renderer from 'react-test-renderer';
import { testRouter } from '../../../utils/testUtils.js';
import Links from '../Links';

describe('<Links />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      testRouter(<Links />)
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
