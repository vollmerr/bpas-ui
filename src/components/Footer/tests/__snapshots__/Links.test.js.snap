// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> renders correctly 1`] = `
<div
  className="sc-bwzfXH dUhRrK"
>
  <a
    className="sc-bxivhb gJUsUs"
    href="http://www.ca.gov/Use"
  >
    Conditions of Use
  </a>
  <a
    className="sc-bxivhb gJUsUs"
    href="http://www.ca.gov/Privacy"
  >
    Privacy Policy
  </a>
  <a
    className="sc-bxivhb gJUsUs"
    href="http://www.ca.gov/Accessibility"
  >
    Accessibility
  </a>
  <a
    className="sc-htpNat kGlyds"
    href="/help"
    onClick={[Function]}
  >
    Contact Us
  </a>
  <div
    className="sc-bdVaJa jHqppW"
  >
    Version 
    test 123
  </div>
</div>
`;
